core/src/main/scala/org/apache/spark/status/StatusJsonHandler.scala

```diff
@@ -0,0 +1,168 @@
+      route
+    }
+  }
+
+  private val noSlash = """[^/]"""
+
+  private val specs: IndexedSeq[(Regex, StatusJsonRoute[_])] = IndexedSeq(
+    "/applications/?".r -> new AllApplicationsJsonRoute(uiRoot),
+    s"/applications/$noSlash+/?".r -> new OneApplicationJsonRoute(uiRoot),
+    s"/applications/$noSlash+/jobs/?".r -> new AllJobsJsonRoute(this),
+    s"/applications/$noSlash+/executors/?".r -> new ExecutorsJsonRoute(this),
+    s"/applications/$noSlash+/stages/?".r -> new AllStagesJsonRoute(this),
+    s"/applications/$noSlash+/stages/$noSlash+/?".r -> new OneStageJsonRoute(this),
+    s"/applications/$noSlash+/storage/rdd/?".r -> new AllRDDJsonRoute(this),
+    s"/applications/$noSlash+/storage/rdd/$noSlash+/?".r -> new RDDJsonRoute(this)
+  )
+
+  private val jsonMapper = {
+    val t = new ObjectMapper()
+    t.registerModule(com.fasterxml.jackson.module.scala.DefaultScalaModule)
+    t.enable(SerializationFeature.INDENT_OUTPUT)
+    t.setSerializationInclusion(JsonInclude.Include.NON_NULL)
+    t
+  }
+
+  val jsonContextHandler = {
+
+    //TODO throw out all the JettyUtils stuff, so I can set the response status code, etc.
+    val servlet = new HttpServlet {
+      override def doGet(request: HttpServletRequest, response: HttpServletResponse) {
+        if (securityManager.checkUIViewPermissions(request.getRemoteUser)) {
+          response.setContentType("text/json;charset=utf-8")
+          route(request) match {
+            case Some(jsonRoute) =>
+              response.setHeader("Cache-Control", "no-cache, no-store, must-revalidate")
+              try {
+                val responseObj = jsonRoute.renderJson(request)
+                val result = jsonMapper.writeValueAsString(responseObj)
+                response.setStatus(HttpServletResponse.SC_OK)
+                response.getWriter.println(result)
+              } catch {
+                case iae: IllegalArgumentException =>
+                  response.setStatus(HttpServletResponse.SC_BAD_REQUEST)
+                  response.getOutputStream.print(iae.getMessage())
+              }
+            case None =>
+              println("no match for path: " + request.getPathInfo)
+              response.setStatus(HttpServletResponse.SC_NOT_FOUND)
+          }
+        } else {
+          response.setStatus(HttpServletResponse.SC_UNAUTHORIZED)
+          response.setHeader("Cache-Control", "no-cache, no-store, must-revalidate")
+          response.sendError(HttpServletResponse.SC_UNAUTHORIZED,
+            "User is not authorized to access this page.")
+        }
+      }
+    }
+    val path = "/json/v1"
+    val contextHandler = new ServletContextHandler
+    val holder = new ServletHolder(servlet)
+    contextHandler.setContextPath(path)
+    contextHandler.addServlet(holder, "/*")
+    contextHandler
+  }
+
+  def withSparkUI[T](request: HttpServletRequest)(f: (SparkUI, HttpServletRequest) => T): T = {
```

Comment:
Can you add a line of Scaladoc to this method?  I guess its use is probably obvious from context, but explicit documentation wouldn't hurt.